src/ui/dialogs/avatarupload.ts

```diff
@@ -0,0 +1,177 @@
+import Dialog from '../Dialog'
+import Client from '../../Client'
+
+let avatarUploadTemplate = require('../../../template/avatarUploadTemplate.hbs');
+
+let dialog: Dialog;
+let base64data : string;
+let mimetype : string;
+
+let height: string;
+let width: string;
+
+let account;
+
+export default function() {
+
+   let content = avatarUploadTemplate();
+
+   dialog = new Dialog(content);
+   let dom = dialog.open();
+
+   let accounts = Client.getAccountManager().getAccounts();
+   for (let i=0;i<accounts.length;i++){
+       let accountoption = $('<option value="'+i+'">'+accounts[i].getUid()+'</option>');
+       dialog.getDom().find('select[name="account"]').append(accountoption);
+   }
+
+   dialog.getDom().find('select[name="account"]').change((e)=>{
+       let val = Number($(e.target).val());
+       account=accounts[val];
+       loadAvatar(account,dialog);
+   });
+
+   console.log(mimetype);
+
+   $('.jsxc-avataruploadinput').change(function(event){
+        let file = (<HTMLInputElement>$('.jsxc-avataruploadinput')[0]).files[0];
+        handleFileSelect(file);
+   });
+
+  account=accounts[0];
+  loadAvatar(account,dialog);
+
+  dom.find('form').submit(onSubmit);
+
+  $('.jsxc-js-clear').on('click',removeAvatar);
+}
+
+function loadAvatar(account,dialog)
+{
+    account.getContact().getVcard().then(function(vcardData) {
+           let data = (vcardData as any);
+           if (data.hasOwnProperty('PHOTO'))
+           {
+              $('.jsxc-avatarimage > img').attr('src',data.PHOTO.src);
+              $('.jsxc-avatarimage > img').css('display','');
+           }
+    });
+}
+
+function handleFileSelect(file) {
+
+  if (!window.File || !window.FileReader || !window.FileList || !window.Blob) {
+    alert('The File APIs are not fully supported in this browser.');
+    return;
+  }
+
+  let fr = new FileReader();
```

Comment:
I think the file reader ist not needed. Should be the same logic as attaching a file (`ChatWindowFileTransferHandler` and `Attachment`).

Reply:
We also use FileReader in Attachement.ts in getDataFromFile()

Reply:
but i see there is a generateThumbnail method we could reuse as util method

Reply:
avatarupload.ts refactored...